pay-api/src/pay_api/services/eft_short_names.py

```diff
@@ -0,0 +1,239 @@
+
+        current_app.logger.debug('>create eft short name')
+        return eft_short_name
+
+    @classmethod
+    def _save(cls, short_name_request: Dict[str, any], short_name: EFTShortnameModel):
+        """Update and save eft short name model."""
+        short_name.short_name = short_name_request.get('shortName')
+        short_name.auth_account_id = short_name_request.get('accountId', None)
+        short_name.flush()
+        short_name.save()
+
+    @classmethod
+    def update(cls, short_name_id: str, short_name_request: Dict[str, Any]) -> EFTShortnames:
+        """Create or update payment account record."""
+        current_app.logger.debug('<update eft short name mapping')
+
+        short_name = EFTShortnameModel.find_by_id(short_name_id)
+        if short_name is None:
+            short_name = EFTShortnameModel()
+
+        EFTShortnames._save(short_name_request, short_name)
+
+        current_app.logger.debug('>update short name mapping')
+        return cls.find_by_short_name_id(short_name.id)
+
+    @classmethod
+    def patch(cls, short_name_id: int, auth_account_id: str) -> EFTShortnames:
+        """Patch eft short name auth account mapping."""
+        current_app.logger.debug('<patch eft short name mapping')
+
+        short_name: EFTShortnameModel = EFTShortnameModel.find_by_id(short_name_id)
+        if auth_account_id is None:
+            raise BusinessException(Error.EFT_SHORT_NAME_ACCOUNT_ID_REQUIRED)
+
+        # If a short name has already been mapped, there could be payments already made.
+        if short_name.auth_account_id is not None:
+            raise BusinessException(Error.EFT_SHORT_NAME_ALREADY_MAPPED)
+
+        short_name.auth_account_id = auth_account_id
+        short_name.save()
+
+        # Update any existing credit mappings with the payment account
+        payment_account = PaymentAccountModel.find_by_auth_account_id(auth_account_id)
+        EFTCreditModel.update_account_by_short_name_id(short_name_id, payment_account.id)
+
+        # Process any invoices owing for short name mapping
+        cls.process_owing_invoices(short_name_id)
+
+        current_app.logger.debug('>patch short name mapping')
+        return cls.find_by_short_name_id(short_name.id)
+
+    @staticmethod
+    def process_owing_invoices(short_name_id: int) -> EFTShortnames:
+        """Process outstanding invoices when an EFT short name is mapped to an auth account id using credits."""
+        current_app.logger.debug('<process_outstanding_invoices')
+        short_name_model: EFTShortnameModel = EFTShortnameModel.find_by_id(short_name_id)
+        auth_account_id = short_name_model.auth_account_id
+
+        # Find invoices to be paid
+        invoices: List[InvoiceModel] = EFTShortnames.get_invoices_owing(auth_account_id)
+        pay_service = PaymentSystemFactory.create_from_payment_method(PaymentMethod.EFT.value)
+
+        if invoices is not None:
```

Comment:
I think this will return empty array? not needed

Reply:
Updated, removed None check